netket/graph/lattice.py

```diff
@@ -104,323 +111,405 @@ def get_edges(atoms_positions, cutoff, distance_atol=cutoff_tol):
         )
         edges = get_true_edges(
-            self._basis_vectors, atoms, cellANDlabel_to_site, extent, distance_atol
+            self._basis_vectors,
+            sites,
+            self._basis_coord_to_site,
+            self.extent,
+            distance_atol,
         )
         graph = _nx.MultiGraph(edges)
 
-        # Rename atoms
-        old_nodes = sorted(set([node for edge in edges for node in edge]))
-        self._atoms = [atoms[old_node] for old_node in old_nodes]
-        self._coord_to_site = {
-            tuple(atom["r_coord"]): new_site
-            for new_site, atom in enumerate(self._atoms)
-        }
+        # Rename sites
+        old_nodes = sorted(set(node for edge in edges for node in edge))
+        self._sites = []
+        for i, (site, _) in enumerate(sites[old_node] for old_node in old_nodes):
+            site.id = i
+            self._sites.append(site)
         new_nodes = {old_node: new_node for new_node, old_node in enumerate(old_nodes)}
         graph = _nx.relabel_nodes(graph, new_nodes)
-
-        self._coords = dicts_to_array(self._atoms, "r_coord")
-        self._lattice_dims = _np.expand_dims(self.extent, 1) * self.basis_vectors
+        self._basis_coord_to_site = {
+            HashableArray(p.cell_coord): p.id for p in self._sites
+        }
+        self._positions = _np.array([p.position for p in self._sites])
+        self._basis_coords = _np.array([p.cell_coord for p in self._sites])
 
         # Order node names
-        nodes = sorted(graph.nodes())
         edges = list(graph.edges())
         graph = _nx.MultiGraph()
-        graph.add_nodes_from(nodes)
+        graph.add_nodes_from([p.id for p in self._sites])
         graph.add_edges_from(edges)
 
-        self._inv_dims = _np.linalg.inv(self._lattice_dims)
-        frac_positions = _np.matmul(self._coords, self._inv_dims) % 1
-        int_positions = _np.around(frac_positions * 10 ** tol_digits).astype(int) % (
-            10 ** tol_digits
-        )
-        self._hash_positions = {
-            HashableArray(element): index for index, element in enumerate(int_positions)
+        lattice_dims = _np.expand_dims(self.extent, 1) * self.basis_vectors
+        self._inv_dims = _np.linalg.inv(lattice_dims)
+        int_positions = self._to_integer_position(self._positions)
+        self._int_position_to_site = {
+            HashableArray(pos): index for index, pos in enumerate(int_positions)
         }
 
         super().__init__(graph)
 
-    @property
-    def basis_vectors(self):
-        return self._basis_vectors
-
-    @property
-    def atoms_coord(self):
-        """
-        Coordinates of atoms in the unit cell.
-        """
-        return self._atoms_coord
-
-    @property
-    def coords(self):
-        """
-        Returns list of coordinates of lattice points
-        """
-
-        return self._coords
+    @staticmethod
+    def _clean_basis(basis_vectors):
```

Comment:
@femtobit the revised version looks good to me, thanks! I am just wondering though what is the purpose of these ```_clean_xxx```functions, maybe you want to add a small comment there in the file?

Reply:
Ah, just internal functions to do the error checking and conversion for some of the init arguments because the code in `__init__` was getting long. I've added brief comments.